src/XmlDocMarkdown.Core/MarkdownGenerator.cs

```diff
@@ -1796,6 +1797,22 @@ private static string GetParameterShortNames(MemberInfo memberInfo)
 			return string.Join(", ", GetParameters(memberInfo).Select(x => RenderTypeName(x.ParameterType.GetTypeInfo())));
 		}
 
+		private static bool IsLessDerived(TypeInfo a, TypeInfo b)
+		{
+			if (!a.IsAssignableFrom(b))
+				return false;
+			if (a.IsGenericType && b.IsGenericType && a.GetGenericTypeDefinition().GetTypeInfo() == b.GetGenericTypeDefinition().GetTypeInfo())
+			{
+				var parameters = a.GetGenericTypeDefinition().GetTypeInfo().GenericTypeParameters;
+				for (int i = 0; i < parameters.Length; i++)
+				{
+					if (!a.GenericTypeArguments[i].GetTypeInfo().IsAssignableFrom(b.GenericTypeArguments[i].GetTypeInfo()))
```

Comment:
Should this only apply to contravariant arguments? Should it be reversed for covariant arguments?

Reply:
Yep. I'll add more test cases.